Search/EventSubscriber/DoctrineOrmIndexRebuildSubscriber.php

```diff
@@ -0,0 +1,136 @@
+use Metadata\ClassMetadata as SearchMetadata;
+use Doctrine\Common\Persistence\Mapping\ClassMetadata as OrmMetadata;
+use Massive\Bundle\SearchBundle\Search\Metadata\IndexMetadata;
+use Massive\Bundle\SearchBundle\Search\Metadata\ClassMetadata;
+
+class DoctrineOrmIndexRebuildSubscriber implements EventSubscriberInterface
+{
+    /**
+     * @var ClassMetadataFactory
+     */
+    private $objectManager;
+
+    /**
+     * @var MetadataFactory
+     */
+    private $searchMetadataFactory;
+
+    /**
+     * @var SearchManager
+     */
+    private $searchManager;
+
+    /**
+     * @var array
+     */
+    private $purged = array();
+
+    /**
+     * @param ObjectManager $objectManager
+     * @param MetadataFactory $searchMetadataFactory
+     * @param SearchManager $searchManager
+     */
+    public function __construct(
+        ObjectManager $objectManager,
+        MetadataFactory $searchMetadataFactory,
+        SearchManager $searchManager
+    )
+    {
+        $this->objectManager = $objectManager;
+        $this->searchMetadataFactory = $searchMetadataFactory;
+        $this->searchManager = $searchManager;
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    public static function getSubscribedEvents()
+    {
+        return array(
+            SearchEvents::INDEX_REBUILD => 'rebuildIndex'
+        );
+    }
+
+    public function rebuildIndex(IndexRebuildEvent $event)
+    {
+        $output = $event->getOutput();
+        $filter = $event->getFilter();
+        $purge = $event->getPurge();
+
+        $metadataFactory = $this->objectManager->getMetadataFactory();
+        $metadata = $metadataFactory->getAllMetadata();
+
+        foreach ($metadata as $class) {
+            if ($filter && !preg_match('{' . $filter . '}', $class->name)) {
+                continue;
+            }
+
+            $searchMeta = $this->searchMetadataFactory->getMetadataForClass($class->getName());
+
+            if (null === $searchMeta) {
+                continue;
+            }
+
+            $classMetadata = $searchMeta->getOutsideClassMetadata();
+
+            if ($purge) {
+                $this->doPurge($output, $classMetadata, $class);
+            }
+
+            $this->rebuildClass($output, $class);
+        }
+    }
+
+    private function doPurge(OutputInterface $output, ClassMetadata $classMetadata, OrmMetadata $ormMetadata)
+    {
+        foreach ($classMetadata->getIndexMetadatas() as $indexMetadata) {
+            $indexName = $indexMetadata->getIndexName();
+
+            if (isset($this->purged[$indexName])) {
+                return;
+            }
+
+            $output->writeln('<info>Purging index</info>: ' . $indexName);
+            $this->searchManager->purge($indexName);
+            $this->purged[$indexName] = true;
+        }
+    }
+
+    private function rebuildClass(OutputInterface $output, OrmMetadata $ormMetadata)
+    {
+        $output->write('<comment>Rebuilding</comment>: ' . $ormMetadata->getName());
```

Comment:
It is maybe not the best strategy to have the listeners write directly to the `OutputInterface`. It would be better for the Command to be responsible for the display of data .. but for now.